tests/test_hamiltonians_trotter.py

```diff
@@ -126,9 +134,14 @@ def test_trotter_hamiltonian_three_qubit_term(backend):
     initial_state = random_statevector(2**4, backend=backend)
     circuit = ham.circuit(dt=dt)
     final_state = backend.execute_circuit(circuit, np.copy(initial_state))
+    if backend.__class__.__name__ == "CupyBackend":
+        mm1 = numpy_backend.cast(mm1.get(), dtype=mm1.dtype)
+        mm2 = numpy_backend.cast(mm2.get(), dtype=mm2.dtype)
+        mm3 = numpy_backend.cast(mm3.get(), dtype=mm3.dtype)
```

Comment:
I believe in the backends we have a `backend.to_numpy()` method to avoid checking for the backend through the name. Would
```suggestion
    mm1 = backend.to_numpy(mm1.get())
    mm2 = backend.to_numpy(mm2.get())
    mm3 = backend.to_numpy(mm3.get())
```
work in this case?

Reply:
This will fail because only the `cupy.ndarray`s have the `.get()` method

Reply:
Sorry, I meant
```suggestion
    mm1 = backend.to_numpy(mm1)
    mm2 = backend.to_numpy(mm2)
    mm3 = backend.to_numpy(mm3)
```
This converts tensors of any backend to numpy arrays, so it will do the `.get()` internally in the `CupyBackend` only, without having to check explicitly with `__class__.__name__`.